clients/src/main/java/org/apache/kafka/clients/consumer/NoOffsetForPartitionException.java

```diff
@@ -32,15 +35,24 @@
     private static final long serialVersionUID = 1L;
 
     private final Set<TopicPartition> partitions;
+    private final Map<TopicPartition, SubscriptionState.FetchState> partitionFetchStates;
 
     public NoOffsetForPartitionException(TopicPartition partition) {
         super("Undefined offset with no reset policy for partition: " + partition);
         this.partitions = Collections.singleton(partition);
+        this.partitionFetchStates = Collections.emptyMap();
     }
 
     public NoOffsetForPartitionException(Collection<TopicPartition> partitions) {
```

Comment:
Since the NoOffsetForPartitionException is only thrown in one place, I think this constructor should be replaced by the new one, and the partitions field should be replaced by the new partitionFetchStates.

Reply:
this exception is being used outside kafka https://jarvis.corp.linkedin.com/codesearch/results?query=%22new+NoOffsetForPartitionException%22&current=1&nresults=10, so don't want to replace it and break the code

Reply:
Thanks for the explanation, that makes sense.